src/lib/operators/table_scan/sorted_segment_search.hpp

```diff
@@ -78,64 +110,85 @@ class SortedSegmentSearch {
+        break;
+      default:
+        Fail("Unsupported predicate condition encountered");
+    }
+
+    if (no_rows_matching) {
+      // _write_rows_to_matches receives _end and iterates until it reaches _end. By setting _begin = _end, it does not
+      // write a PosList.
+      _begin = _end;
+    }
+
+    if (all_rows_matching || no_rows_matching) {
       return;
     }
 
-    // clang-format off
-    if (_is_ascending) {
-      switch (_predicate_condition) {
-        case PredicateCondition::GreaterThanEquals:
-          _begin = _get_first_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::GreaterThan:
-          _begin = _get_last_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::LessThanEquals:
-          _end = _get_last_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::LessThan:
-          _end = _get_first_bound(_first_search_value, _begin, _end);
-          return;
-        default: Fail("Unsupported predicate condition encountered");
-      }
-    } else {
-      switch (_predicate_condition) {
-        case PredicateCondition::LessThanEquals:
-          _begin = _get_first_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::LessThan:
-          _begin = _get_last_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::GreaterThanEquals:
-          _end = _get_last_bound(_first_search_value, _begin, _end);
-          return;
-        case PredicateCondition::GreaterThan:
-          _end = _get_first_bound(_first_search_value, _begin, _end);
-          return;
-        default: Fail("Unsupported predicate condition encountered");
-      }
+    switch (predicate_condition) {
+      case PredicateCondition::Equals:
+        _begin = _get_first_bound(_first_search_value, _begin, _end);
+        _end = _get_last_bound(_first_search_value, _begin, _end);
+        return;
+      case PredicateCondition::GreaterThanEquals:
+        _begin = _get_first_bound(_first_search_value, _begin, _end);
+        return;
+      case PredicateCondition::GreaterThan:
+        _begin = _get_last_bound(_first_search_value, _begin, _end);
+        return;
+      case PredicateCondition::LessThanEquals:
+        _end = _get_last_bound(_first_search_value, _begin, _end);
+        return;
+      case PredicateCondition::LessThan:
+        _end = _get_first_bound(_first_search_value, _begin, _end);
+        return;
+      default:
+        Fail("Unsupported predicate condition encountered");
     }
-    // clang-format on
   }
 
   // This function sets the offset(s) that delimit the result set based on the predicate condition and the sort order
   void _set_begin_and_end_positions_for_between_scan() {
     DebugAssert(_second_search_value, "Second Search Value must be set for between scan");
```

Comment:
Maybe this should be asserted in release builds, too.

Reply:
We call `_set_begin_and_end_positions...` internally depending on the existence of `_second_search_value`. As this method cannot be called externally without that parameter, I don't see a reason for an `Assert` tbh,